src/Symfony/Component/Validator/Validator/LegacyValidator.php

```diff
@@ -69,7 +69,7 @@ public function getMetadataFactory()
 
     private static function testConstraints($constraints)
     {
-        return null === $constraints || $constraints instanceof Constraint || (is_array($constraints) && current($constraints) instanceof Constraint);
+        return null === $constraints || $constraints instanceof Constraint || (is_array($constraints) && (empty($constraints) || current($constraints) instanceof Constraint));
```

Comment:
I know it's micro optimization, but ``` empty ``` seems to have better performance http://webjawns.com/2014/04/php-benchmark-count-vs-empty/. Anyway I changed it to ``` count ``` ;)